### PR TITLE
docs: add OpenBSD rc.d script

### PR DESCRIPTION
**Pull request by skarekrow**
This has a few prerequisites:
- Creating a `_cloudflare_ddns` user
- Add a `cloudflare_ddns` section to your `/etc/login.conf` (for env variables, this is noted in the accompanying `README` commit

Relies on #733

### CODE REVIEW

**Comment by skarekrow** [comment]
@favonia I tried to keep the style of your README, feel free to edit if you want it laid out differently. I also named the directory in the repo as `rc.d`, but it could just as easily be `init`, I'll leave that up to you. I think most non-systemd init systems use `rc.d` as the naming, but I'm not 100%

**Comment by favonia** [comment]
@skarekrow Thank you! Could you put the files under `contrib/` because, sadly, I will not be able to personally maintain these files? Feel free to update the README and/or rearrange files there.

**Comment by skarekrow** [comment]
@favonia Whoops didn't even see the `contrib` directory! I moved the systemd services to have a similar structure and updated the README's for both.

File: contrib/README.markdown [hunk 1]
```diff
@@ -6,5 +6,24 @@ _(contributed by [Thomas @sumgryph](https://github.com/symgryph))_
...
-- See [cloudflare-ddns.service.env](./cloudflare-ddns.service.env) for a sample systemd environment file.
+- See [cloudflare-ddns.service](./linux/cloudflare-ddns.service) for a sample systemd service unit file.
+- See [cloudflare-ddns.service.env](./linux/cloudflare-ddns.service.env) for a sample systemd environment file.
+
+## 🐡 OpenBSD
+
+_(contributed by [Brandon @skarekrow](https://github.com/skarekrow))_
+
+To use:
+- Simply copy the [rc.d](./openbsd/cloudflare_ddns) script located in the repo
```
Comment (favonia) [inline contrib/README.markdown:17]:
```suggestion
- Copy the shipped [rc.d script](./openbsd/cloudflare_ddns) into `/etc/rc.d/`
```

**Comment by favonia** [comment]
@skarekrow Are you still interested in updating this PR? Please let me know. Another relevant issue is #952 where I'm relicensing this project under Apache 2.0 _with LLVM exceptions,_ and I will need your approval here. You technically only agreed to license this PR under the current license (Apache 2.0 _without_ LLVM exceptions).

File: contrib/README.markdown [hunk 2]
```diff
@@ -6,5 +6,24 @@ _(contributed by [Thomas @sumgryph](https://github.com/symgryph))_
...
+- The rc.d script assumes a user called `_cloudflare_ddns` will be used. This is easily created by doing `useradd -s /sbin/nologin -d /var/empty _cloudflare_ddns`
+- Create a `/etc/login.conf` entry for the daemon specifying the environment variables you wish to use:
+
+```sh
+cloudflare_ddns:\
+        :setenv=CF_API_TOKEN=YOUR_TOKEN,DOMAINS=THE_DOMAINS_YOU_WISH_TO_USE,EMOJI=false:\
```
[the remark this answers is not in the document]
Comment (skarekrow) [reply contrib/README.markdown:23]: That's fair! I honestly only ever tested with one domain as that's all I use that's updated. I would think you may want to escape the commas perhaps?

**Comment by skarekrow** [comment]
@favonia Sorry about that! I missed your comment and just saw the merges. I'm OK with whatever license you choose for this project, my contribution is minor at best!

File: contrib/README.markdown (hunk 2 above)
[the remark this answers is not in the document]
Comment (favonia) [reply contrib/README.markdown:23]: @skarekrow I checked the documentation and it doesn't mention any way to escape commas...

[the remark this answers is not in the document]
Comment (skarekrow) [reply contrib/README.markdown:23]: @favonia Yeah this doesn't seem documented well. I was looking at the man pages trying to figure out what the expected process is for that. The answer seems to lie in [here](https://man.openbsd.org/cgetent.3) but I cannot figure out the arcane invocation for it :D

[the remark this answers is not in the document]
Comment (favonia) [reply contrib/README.markdown:23]: Wow, do you mean that `\054` will work? (I don't use OpenBSD.)

[the remark this answers is not in the document]
Comment (skarekrow) [reply contrib/README.markdown:23]: Haha honestly I don't know. It's gonna be a little bit before I have bandwidth to test this and update the docs with what I find. Maybe we add a small blurb saying we haven't tested using multiple domains via this method and then have another PR to update with the findings?

[the remark this answers is not in the document]
Comment (favonia) [reply contrib/README.markdown:23]: @skarekrow Are there other reasons not to accept my suggestion, then?

[the remark this answers is not in the document]
Comment (skarekrow) [reply contrib/README.markdown]: @favonia Just me apparently not committing it lol. I think it's fine!

**Pull request merged**